File: CarCareHub_/CarCareHub.Model/Zaposlenik.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Model
{
    public class Zaposlenik
    {
        public int ZaposlenikId { get; set; }
        public string? Ime { get; set; }
        public string? Prezime { get; set; }
        public string? mb { get; set; }
        public string? BrojTelefona { get; set; }
        public Grad? Grad { get; set; }
        public int? GradId { get; set; }
        public bool Vidljivo { get; set; }
        public DateTime? DatumRodjenja { get; set; }
        public string? Adresa { get; set; }
        public string? Email { get; set; }
        public string? Username { get; set; }
        public string? LozinkaSalt { get; set; }
        public string? LozinkaHash { get; set; }
        public string? Password { get; set; }
        public int? UlogaId { get; set; }
        public virtual Uloge? Uloga { get; set; }
        public int? AutoservisId { get; set; }
        public virtual Autoservis? Autoservis { get; set; }
        public int? FirmaAutodijelovaId { get; set; }
        public virtual FirmaAutodijelova? FirmaAutodijelova { get; set; }
    }

}

File: CarCareHub_/UI_/flutter_desktop/lib/models/zaposlenik.dart
// ignore_for_file: non_constant_identifier_names

import 'package:flutter_mobile/models/autoservis.dart';

import 'package:flutter_mobile/models/firmaautodijelova.dart';
import 'package:flutter_mobile/models/grad.dart';
import 'package:flutter_mobile/models/uloge.dart';
import 'package:json_annotation/json_annotation.dart';

part 'zaposlenik.g.dart';

@JsonSerializable()
class Zaposlenik {

  int? zaposlenikId;
  String? ime;
  String? prezime;
  String? mb;
  String? brojTelefona;
  Grad? grad;
  int? gradId;
  DateTime? datumRodjenja;
  String? email;
  String? username;
  String? password;
  String? PasswordAgain;
bool? vidljivo;
  String? lozinkaSalt;
  String? lozinkaHash;
  int? autoservisId;
  Autoservis? autoservis;
  int? firmaAutodijelovaId;
  FirmaAutodijelova? firmaAutodijelova;
  int? ulogaId;
  Uloge? uloga;
  String? passwordAgain;

  String? adresa;

 
  Zaposlenik({
    this.zaposlenikId,
    this.vidljivo,
    this.ime,
    this.prezime,
   this.mb, 
    this.brojTelefona,
    this.gradId,
    this.datumRodjenja,
    this.email,
    this.username,
    this.lozinkaSalt,
    this.lozinkaHash,
    this.password,
    this.ulogaId,
    this.autoservisId,
    this.firmaAutodijelovaId,
    this.grad,
    this.autoservis,
    this.firmaAutodijelova,
    this.uloga,
    this.passwordAgain,
    this.adresa
    
  });

  factory Zaposlenik.fromJson(Map<String, dynamic> json) =>
      _$ZaposlenikFromJson(json);

  Map<String, dynamic> toJson() => _$ZaposlenikToJson(this);
}
